### PR TITLE
Introduce config schema for Intel Gaudi

### PR DESCRIPTION
**Pull request by alinaryan**
Adds a Intel Gaudi 3 config schema to the CLI.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    max_ctx_size: 4096
+  # Path to model that will be served for inference
+  model_path: ~/.cache/instructlab/models/instructlab/granite-7b-lab
+  # vLLM serving settings
+  vllm:
+    gpus: 4
+    # the family of model being served - used to determine the appropriate chat template
+    llm_family: ''
+    # additional arguments to be supplied directly to vLLM
+    vllm_args: ["--tensor-parallel-size", "4"]
```
Comment (cdoern) [inline src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml:105]:
```suggestion
    vllm_args: ["--tensor-parallel-size", "8"]
```

Comment (cdoern) [reply src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml:105]: or is 8 GPUS on SDG still broken?

**Review by cdoern — COMMENTED** [review]
one comment I am not sure about

File: src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml (hunk 1 above)
Comment (alinaryan) [reply src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml]: Spoke with Aakanksha, 8 GPUs should be fine! Updated

File: src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  # Teacher model specific settings
+  teacher:
+    # Serving backend to use to host the teacher model
+    backend: vllm
+    # Path to teacher model that will be used to synthetically generate training data
+    model_path: ~/.cache/instructlab/models/mistralai/Mixtral-8x7B-Instruct-v0.1
+    # vLLM serving settings
+    vllm:
+      # number of GPUs to allocate to vLLM
+      gpus: 4
```
Comment (cdoern) [inline src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml:78]:
```suggestion
      gpus: 8
```

File: src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  visible_overflow: true
+evaluate:
+  # Base taxonomy branch
+  base_branch: null
+  # Directory where the model to be evaluated is stored
+  base_model: ~/.cache/instructlab/models/instructlab/granite-7b-lab
+  # Taxonomy branch containing custom skills/knowledge that should be used for evaluation runs
+  branch: null
+  # Number of GPUs to use for running evaluation
+  gpus: 4
```
Comment (cdoern) [inline src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml:24]:
```suggestion
  gpus: 8
```

File: src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    gpu_layers: -1
+    # the family of model being served - used to determine the appropriate chat template
+    llm_family: ''
+    # maximum number of tokens that can be processed by the model
+    max_ctx_size: 4096
+  # Path to model that will be served for inference
+  model_path: ~/.cache/instructlab/models/instructlab/granite-7b-lab
+  # vLLM serving settings
+  vllm:
+    gpus: 4
```
Comment (cdoern) [inline src/instructlab/profiles/intel/gaudi/mi300x/mi300x_x8.yaml:101]:
```suggestion
    gpus: 8
```

**Review by cdoern — COMMENTED** [review]
found a couple more.

Can we wait until auto-detection merges so then we can add a unit test which auto-detects this? If not, thats fine too

**Review by cdoern — COMMENTED** [review]
sorry i just realized, the file name shouldn't have mi300x as that is AMD. this should be `profiles/intel/gauid/gaudi_3.yaml`

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**